github4s/src/main/scala/github4s/Github.scala

```diff
@@ -16,55 +16,36 @@
 
 package github4s
 
-import cats.data.{EitherT, Kleisli}
-import cats.MonadError
-import cats.implicits._
-import github4s.GithubResponses._
-import github4s.free.interpreters.Interpreters
-
-import scala.concurrent.Future
-
-/**
- * Represent the Github API wrapper
- * @param accessToken to identify the authenticated user
- */
-class Github(accessToken: Option[String] = None) {
-
-  lazy val users         = new GHUsers(accessToken)
-  lazy val repos         = new GHRepos(accessToken)
-  lazy val auth          = new GHAuth
-  lazy val gists         = new GHGists(accessToken)
-  lazy val issues        = new GHIssues(accessToken)
-  lazy val activities    = new GHActivities(accessToken)
-  lazy val gitData       = new GHGitData(accessToken)
-  lazy val pullRequests  = new GHPullRequests(accessToken)
-  lazy val organizations = new GHOrganizations(accessToken)
+import java.util.concurrent.TimeUnit.MILLISECONDS
+import cats.effect.ConcurrentEffect
+import github4s.algebras._
+import github4s.modules._
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.Duration
+
+class Github[F[_]: ConcurrentEffect](accessToken: Option[String], timeout: Option[Duration])(
+    implicit ec: ExecutionContext) {
+
+  val module: GithubAPIs[F] =
+    new GithubAPIv3[F](accessToken, timeout.getOrElse(Duration(1000l, MILLISECONDS)))
+
+  val users: Users[F]                 = module.users
+  val repos: Repositories[F]          = module.repos
+  val auth: Auth[F]                   = module.auth
+  val gists: Gists[F]                 = module.gists
+  val issues: Issues[F]               = module.issues
+  val activities: Activities[F]       = module.activities
+  val gitData: GitData[F]             = module.gitData
+  val pullRequests: PullRequests[F]   = module.pullRequests
+  val organizations: Organizations[F] = module.organizations
```

Comment:
Maybe it's not a big deal, but we also might want to make all these modules instantiations as `lazy`?